### DISCUSSION TITLE
Importing .mtl files for .obj files

### DISCUSSION THREAD

**Discussion by huynmela** [JuliaPlots/Makie.jl#1607] [Q&A] [answered]
Hello! 

I am attempting to load in a .obj file with its associated textures (.jpeg and .mtl). I am also trying to view this .obj file in 3D using WGLMakie. I have two questions:
1. An issue I am running into when using WGLMakie to view my .obj file with a matcap texture is that the texture does not appear-- instead, the .obj appears completely black (no lighting?). Is it possible to apply matcaps while using WGLMakie?
2. Without WGLMakie, the matcap applys, but in the incorrect areas. I haven't found a way in the docs or online to load in the associated .mtl file with the .obj and .jpeg. Can Makie load in .mtl files? 

Below is my sample code. Thanks!

```
using FileIO, GLMakie, ImageMagick, JSServe, WGLMakie

texture_pic = load(assetpath(joinpath(pwd(), "data", "xx.jpg")))
obj_mesh = load(assetpath(joinpath(pwd(), "data", "xx.obj")))

mesh(
    obj_mesh, matcap=texture_pic,
    figure = (resolution = (700, 1000),)
)
```

**Comment by ffreyer** [comment, marked as answer]
1. WGLMakie doesn't have matcaps so it just uses whatever color is set.
2. Maybe you're trying to render a normal texture as a matcap? Try `color = texture_pic` instead. (Matcaps are textures that use normals to map color onto a mesh. Afaik that usually only includes front facing normals so there isn't much transforming you can do.) The loading is done by FileIO/MeshIO. I'm not aware of any mtl loading capabilities. Makie also doesn't have a clean concept of materials yet (well maybe RPRMakie does)

**Comment by SimonDanisch** [comment]
We implemented a mtl parser ages ago, but it has been bit-rotting away: https://github.com/JuliaIO/WavefrontObj.jl/blob/master/src/mtl.jl..
Someone would need to revive that code...

Do you really want matcap (pretty much an environment map, used for more interesting reflections), or do you just want a textured mesh?

**Comment by huynmela** [comment]
@ffreyer @SimonDanisch 
I see! My issue was thinking matcaps acted the same as a textured mesh. Changing it to `color=texture_pic` worked perfectly. Much appreciation to you both, thank you!
